File: chatwoot/kotlin/src/main/kotlin/oseg/chatwoot_examples/UpdateAnIntegrationsHookExample.kt
package oseg.chatwoot_examples

import com.chatwoot.client.infrastructure.*
import com.chatwoot.client.apis.*
import com.chatwoot.client.models.*

import java.io.File
import java.time.LocalDate
import java.time.OffsetDateTime
import kotlin.collections.ArrayList
import kotlin.collections.List
import kotlin.collections.Map
import com.squareup.moshi.adapter

@ExperimentalStdlibApi
class UpdateAnIntegrationsHookExample
{
    fun updateAnIntegrationsHook()
    {
        ApiClient.apiKey["userApiKey"] = "USER_API_KEY"
        // ApiClient.apiKey["agentBotApiKey"] = "AGENT_BOT_API_KEY"
        // ApiClient.apiKey["platformAppApiKey"] = "PLATFORM_APP_API_KEY"

        val integrationsHookUpdatePayload = IntegrationsHookUpdatePayload()
        )

        try
        {
            val response = IntegrationsApi().updateAnIntegrationsHook(
                accountId = null,
                hookId = null,
                _data = integrationsHookUpdatePayload,
            )

            println(response)
        } catch (e: ClientException) {
            println("4xx response calling IntegrationsApi#updateAnIntegrationsHook")
            e.printStackTrace()
        } catch (e: ServerException) {
            println("5xx response calling IntegrationsApi#updateAnIntegrationsHook")
            e.printStackTrace()
        }
    }
}

File: chatwoot/typescript-node/src/UpdateAnIntegrationsHookExample.ts
import * as fs from 'fs';
import api from "chatwoot_client"
import models from "chatwoot_client"

const apiCaller = new api.IntegrationsApi();
apiCaller.setApiKey(api.IntegrationsApiApiKeys.userApiKey, "USER_API_KEY");
// apiCaller.setApiKey(api.IntegrationsApiApiKeys.agentBotApiKey, "AGENT_BOT_API_KEY");
// apiCaller.setApiKey(api.IntegrationsApiApiKeys.platformAppApiKey, "PLATFORM_APP_API_KEY");

const integrationsHookUpdatePayload: models.IntegrationsHookUpdatePayload = {
};

apiCaller.updateAnIntegrationsHook(
  0, // accountId
  0, // hookId
  integrationsHookUpdatePayload, // data
).then(response => {
  console.log(response.body);
}).catch(error => {
  console.log("Exception when calling IntegrationsApi#updateAnIntegrationsHook:");
  console.log(error.body);
});
